core/templates/dev/head/pages/collection_editor/collection_editor.html

```diff
@@ -91,6 +91,8 @@
   <script src="/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js"></script>
   <script src="/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js"></script>
   <script src="/templates/dev/head/domain/exploration/ContentIdsToAudioTranslationsObjectFactory.js"></script>
+  <script src="/templates/dev/head/domain/exploration/WrittenTranslationObjectFactory.js"></script>
+  <script src="/templates/dev/head/domain/exploration/WrittenTranslationsObjectFactory.js"></script>
```

Comment:
I have a feeling that this naming is going to be very confusing (in the future). Could you elaborate what these two files are used for? (Is it just singular and plural as the names suggest?) Maybe we can find better names here!

Reply:
@nithusha21, We have a similar naming conviction for state and states object. Also, written translation is an Object representing a text translation whereas written translations contain a translation mapping which use to map a contentId with writtent_translations for a specific language. 

Also, FYI we have a similar name in the backend. It would be great if you can suggest a good name for this (I tried my best and this is what I was able to finalize)

Reply:
"writtent_translations for a specific language" did you mean singular or plural? (As per current naming it would relate to 2 different classes!

I actually have no good suggestions here, but as illustrated above, it can get really confusing if you string these together in a sentence... Maybe something along the lines of language_to_written_translation_mapping (maybe to large) would be more informative?

Reply:
@nithusha21 I've added a dict form of written_translations belwo:


```
written_translations = {
    translations_mapping: {
        content_id_1: { langauge_code1: written_translation, langauge_code_2: written_translation... }

        content_id_2: {.....}
    }
}
```

Also written translation:

written_translation: {
    html: str.
    need_update: boolean.
}

Reply:
oh ok.. This is slightly confusing overall, but I think it is ok to leave it as is for now (considering similar patterns elsewhere). 

maybe the top level object can be called written_translations_for_state? No strong opinion here. Leaving the decision to you :)

Reply:
Written_translations are the inner properties of the state. (Currently, I'm leaving it as is.) Thanks! :)